text/0000-intl-v2.md

```diff
@@ -0,0 +1,218 @@
+
+```json
+{
+  "n":{
+    "01ixx0000005ztp":{
+      "t":"n",
+      "l":"01Ixx0000005ZTP",
+      "s":"c",
+      "v":{
+        "0":"CO1",
+        "1":"CO1s"}
+    },
+    "account":{
+      "t":"n",
+      "l":"account",
+      "s":"c",
+      "v":{
+        "0":"Company",
+        "1":"Companies"}
+    }
+  },
+  "a":{},
+  "d":{}
+}
+```
+
+Knowing that we can retrieve object metadata, as well as, nouns, articles, adjectives, etc. We can then run it through a custom parser that transforms all of these into Fluent messages:
+
+```fluent
+element 
+   .content = { $accounts ->
+       [one] {"A"}
+       *[other] {""}
+    } { $accounts ->
+      [one] Company
+      *[other] Companies
+    }
+}.
+```
+
+## Pros and Cons
+
+Pros:
+    * Benefits to Developer Productivity
+        * Being able to access labels as data, rather than something that requires further computing at runtime.
+        * Being able to rely on an open source and standard solution, like fluent, allow us to get to market quicker with a well stablished solution.
+        * Being able to use labels declaratively. 
+    * Benefits to Performance/Non-functional requirements 
+        * No noted performance implications at first sight, all i18n specific data to be provisioned as part of the module graph for a component.
+        * Not allowing ad-hoc fetch operations for retrieving dynamic entity data.
+    * Service level expectations - Performance/Availability
+        * No additional service is required for this to function.
+    * Localization
+        * Gramatticus, our well stablished legacy platform for localization, will remain as the service that powers the translations.
+
+Cons:
+
+    * Complexity:
+        * LWC Compiler must be revamped.
+        * New language to be integrated with code editors, and linters to be introduced.
+        * New metadata to be added to LWC Compilation to describe the label and entities dependencies.
+    * External Dependency:
+        * Using fluent, as the primary language to describe messages is tricky because we don't control the destiny of that project.
+
+## Adoption strategy
+
+### Backward Compatibility & Risks
+
+One of the compelling things about this proposal is that it does not affect in any way the existing HTML Parser, or syntax. It does not introduce any new directive either. In previous attempts to solve this problem, that was one of the major concerns. This new feature should be backward compatibility, and users of the framework can continue using the existing label resolution in platform alongside the new feature.
+
+### Prior Art
+
+FormatJS, which relies of helpers and template literals to define the messages, and format the data. Our own implementation of the label system today, which promotes the import of a platform label, and the manual transformation of such string value to produce the final result. As a result, that component is just not localizable anymore.
```

Comment:
It would be great to expand this section with examples from other popular frameworks (React, Vue, Svelte, Angular, etc.) and meta frameworks (Next, Nuxt, SvelteKit, Remix, etc.). How do they handle i18n?

Reply:
I can certainly add more details, formatjs website does have a good set of examples for some of those frameworks.